src/did/src/sign_strategy.rs

```diff
@@ -0,0 +1,334 @@
+use std::borrow::Cow;
+use std::cell::RefCell;
+
+use async_trait::async_trait;
+use candid::CandidType;
+use eth_signer::ic_sign::{IcSigner, SigningKeyId};
+use eth_signer::{Signer, Wallet};
+use ethers_core::k256::ecdsa::SigningKey;
+use ethers_core::types::transaction::eip2718::TypedTransaction;
+use ethers_core::utils;
+use ic_exports::ic_ic00_types::DerivationPath;
+use ic_stable_structures::Storable;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+use crate::error::{EvmError, Result};
+use crate::transaction::Signature;
+use crate::H160;
+
+/// A trait that abstracts out the transaction signing component
+#[async_trait(?Send)]
+pub trait TransactionSigner {
```

Comment:
It seems to me that this trait is not needed because the `TxSigner` is an enum

Reply:
I would personally leave the trait:
1. It unifies the implementation for different signing strategies
2. At the component that actually stores the strategy we have a getter method that returns `impl TxSigner` not exposing the concrete type